packages/slice-machine/src/components/SideNav/SideNav.tsx

```diff
@@ -132,6 +132,38 @@ export const SideNavLink: FC<SideNavLinkProps> = ({
   );
 };
 
+export type SideNavButtonProps = {
+  title: string;
+  active?: boolean;
+  Icon: FC<SVGProps<SVGSVGElement>>;
+  RightElement?: ReactNode;
+  component?: "button" | FC<{ onClick?: (event: MouseEvent) => void }>;
+  onClick?: (event: MouseEvent) => void;
+};
+
+export const SideNavButton: FC<SideNavButtonProps> = ({
+  title,
+  RightElement,
+  Icon,
+  active,
+  component: Comp = "button",
```

Comment:
⚠️ Let's not accept a component, today, if it's only a button and always a button, we don't need it.
Simplify a lot the code.

Reply:
Makes sense, done ✅